File: DSA-CP/CodeForces/Class_DSA/Sorting.py
import numpy as np

def insertion_sort(arr):
    for i in range(len(arr)):
        temp = arr[i]
        ite=0
        print('{',end='')
        for j in range(i,0,-1):
            ite=j
            print(j,end=',')
            if temp<arr[j]:
                arr[j]=arr[j-1]
        arr[ite]=temp

        print('}',i,arr)
    print(arr)  
x=np.random.randint(low=0,high=10,size=10)
insertion_sort(arr=x)

File: DSA-CP/CodeForces/Class_DSA/Sorting.cpp
#include<bits/stdc++.h>

using namespace std;


int main(){
    int arr[]={9,8,7,6,5,4,3,2,10};
    for(int i=1;i<9;i++){
        int temp=arr[i];int j;
        for(j=i-1;j>=0;j--){
            if(arr[j]>temp){
                arr[j+1]=arr[j];
            }
            else {
                break;
            }
        }
        arr[j+1]=temp;
    }
    for(int i=0;i<9;i++) cout<<arr[i]<<" ";
    return 0;
}